services/keyword_service/keyword_test.go

```diff
@@ -87,6 +87,47 @@ func (s *KeywordServiceDbTestSuite) TestGetAllWithInvalidUser() {
 	assert.Equal(s.T(), nil, err)
 }
 
+func (s *KeywordServiceDbTestSuite) TestGetKeywordResultWithValidKeywordAndUser() {
+	keyword := models.Keyword{UserID: s.userID, Keyword: faker.Name()}
+	db.GetDB().Create(&keyword)
+
+	result, err := s.keywordService.GetKeywordResult(keyword.ID)
+
+	assert.Equal(s.T(), keyword.Keyword, result.Keyword)
+	assert.Equal(s.T(), nil, err)
+}
+
+func (s *KeywordServiceDbTestSuite) TestGetKeywordResultWithValidKeywordButInvalidUser() {
+	keyword := models.Keyword{UserID: s.userID, Keyword: faker.Name()}
+	db.GetDB().Create(&keyword)
+
+	keywordService := KeywordService{CurrentUserID: 999999}
+	result, err := keywordService.GetKeywordResult(keyword.ID)
+
+	assert.Equal(s.T(), models.Keyword{}, result)
+	assert.Equal(s.T(), "keyword not found", err.Error())
+}
+
+func (s *KeywordServiceDbTestSuite) TestGetKeywordResultWithWrongKeywordTypeButValidUser() {
+	keyword := models.Keyword{UserID: s.userID, Keyword: faker.Name()}
+	db.GetDB().Create(&keyword)
+
+	result, err := s.keywordService.GetKeywordResult("invalidKeyword")
+
+	assert.Equal(s.T(), models.Keyword{}, result)
+	assert.Equal(s.T(), "invalid input", err.Error())
+}
+
+func (s *KeywordServiceDbTestSuite) TestGetKeywordResultWithInvalidKeywordButValidUser() {
+	keyword := models.Keyword{UserID: s.userID, Keyword: faker.Name()}
+	db.GetDB().Create(&keyword)
+
+	result, err := s.keywordService.GetKeywordResult("999999")
```

Comment:
@gutakk Would the result be the same if we change `"999999"` to integer `999999`? 🤔 
Also how about changing the function name to `TestGetKeywordResultWithInvalidKeywordIDButValidUser` because we are passing the invalid keyword ID.

Reply:
Good catch for the function name. Fixed in 641accbf6d741e60b2c80e4583ed186f9e148c22

The `GetKeywordResult` parameters type is `interface{}` and it can take both string and integer and the result is the same. I added more test case about it in 3c4c3f622a6f2b5747b140097012b9bfbab3a81b